### PR TITLE
Increase performance of STDOUT JSON adapter

### PR DESCRIPTION
**Pull request by take-five**
1. Don't serialize timestamps in ruby, let OJ handle it
2. Don't stringify log level, let OJ convert it to string

Before:
```
$ time ruby perf.rb > /dev/null

real	0m2.550s
user	0m2.474s
sys	0m0.061s
```

After:
```
$ time ruby perf.rb > /dev/null

real	0m1.679s
user	0m1.616s
sys	0m0.052s
```

perf.rb:
```
require 'bundler/setup'
Bundler.require
logger = Logasm.build('test_service', stdout: {level: 'info', json: true})

100_000.times do
  logger.info "hello there", a: "asdf", b: 'eadsfasdf', c: {hello: 'there'}
end
```

### CODE REVIEW

File: lib/logasm/utils.rb [hunk 1]
```diff
@@ -32,28 +31,6 @@ def self.application_name(service_name)
       underscore(service_name)
     end
 
-    def self.overwritable_params
-      {
-        :@timestamp => Time.now.utc.iso8601(DECIMAL_FRACTION_OF_SECOND)
```
Comment (urmastalimaa) [inline lib/logasm/utils.rb]: constant unused now

File: spec/utils_spec.rb [hunk 2]
```diff
@@ -41,13 +42,5 @@
         expect(subject[:host]).to eq('xyz')
       end
     end
-
-    context 'when time object in metadata' do
-      let(:metadata) { {time: Time.utc(2016, 1, 5, 10, 38)} }
```
Comment (urmastalimaa) [inline spec/utils_spec.rb]: Why is this removed?

Comment (urmastalimaa) [reply spec/utils_spec.rb]: ah, nvm

File: spec/utils_spec.rb [hunk 3]
```diff
@@ -2,26 +2,27 @@
...
       expect(event[:application]).to eq(application_name)
     end
 
     it 'includes log level' do
-      expect(event[:level]).to eq('info')
+      expect(event[:level]).to eq(:info)
     end
 
     it 'includes timestamp' do
-      expect(event[:@timestamp]).to eq('2015-10-11T23:10:21.123Z')
```
Comment (urmastalimaa) [inline spec/utils_spec.rb]: Can we have integration tests that include serialization? Now that serialization is not directly tested anymore.

File: lib/logasm/adapters/stdout_json_adapter.rb [hunk 4]
```diff
@@ -1,6 +1,12 @@
 class Logasm
   module Adapters
     class StdoutJsonAdapter
+      OJ_DUMP_OPTS = {
+        mode: :compat,
+        time_format: :xmlschema,
```
Comment (urmastalimaa) [inline lib/logasm/adapters/stdout_json_adapter.rb]: Can you explain time_format paramete change from :ruby to :xmlschema?

Comment (take-five) [reply lib/logasm/adapters/stdout_json_adapter.rb]: Previously we used `time.iso8601` to convert time to string. Oj can do it more effectively. `time_format: :xmlschema` does exactly this - converts all time objects to strings in required format

**Comment by indrekj** [comment]
```
irb(main):013:0> puts Oj.dump({x: Time.now.utc},  time_format: :xmlschema, mode: :wab)
{"x":"2018-03-02T11:20:23.865997000Z"}
```
`mode: :wab` seems to work but I'm not sure what this mode is.

I'm using oj 3.4.0

**Comment by indrekj** [comment]
`time_format` also seems to work with `mode: :custom`

```
puts Oj.dump({x: Time.now.utc},  mode: :custom, time_format: :xmlschema)
{"x":"2018-03-02T11:26:50.176158000Z"}
```

https://github.com/ohler55/oj/blob/ed47a3511c6ddac958113bc69517abff9e1dce98/pages/Modes.md

**Comment by take-five** [comment]
Yeah, mode `:compat` doesn't support `:time_format` option, which is very sad
